apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/sse/SseTransportMessageFactory.kt

```diff
@@ -0,0 +1,28 @@
+package com.apollographql.apollo3.network.sse
+
+/**
+ * Creates client requests
+ */
+open class SseTransportMessageFactory(private val messageType: SseTransportMessage.MessageType = SseTransportMessage.MessageType()) {
```

Comment:
Why do we need a Factory at this point? Maybe this could be simplified. Also I find it unexpected that it has a mutable `type`.

Reply:
I am anticipating a bit of friction on the strings that are held by `MessageType` when I try to reconcile EnisDenjo's spec with what we have live inside Woolworths. Keen to explore other ways to decouple it.

About the mutable var, I'll look into how to write `Builder` pattern with val. Maybe passing mandatory parameter in constructor?

Reply:
Yes I believe a Builder taking arguments would make sense.